packages/xforms-engine/src/client/TextRange.ts

```diff
@@ -149,11 +150,13 @@ export type TextOrigin =
  * {@link https://getodk.github.io/xforms-spec/#languages | translation}).
  */
 export interface TextRange<Role extends TextRole, Origin extends TextOrigin = TextOrigin> {
+	readonly evaluator: EngineXPathEvaluator | null; // TODO: can it be null?
 	readonly origin: Origin;
 	readonly role: Role;
 
 	[Symbol.iterator](): Iterable<TextChunk>;
 
 	get asString(): string;
 	get formatted(): unknown;
+	get image(): Promise<MediaResource> | null;
```

Comment:
Ultimately, it seems simple and easy to understand, so I didn't add extra classes or interfaces for media.

Reply:
Can we avoid the null? Maybe do a rejected promise instead?